src/Trace/Trace.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+/**
+ * This plain PHP class represents a Trace resource. Traces belong to a
+ * project and have many TraceSpans. See:
+ * https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#resource-trace
+ *
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder();
+ * $traceClient = $cloud->trace();
+ *
+ * $trace = $traceClient->trace();
+ * ```
+ */
+class Trace
+{
+    /**
+     * The id of the project this trace belongs to.
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     * @var string
+     */
+    private $traceId;
+
+    /**
+     * List of TraceSpans to report
+     * @var TraceSpan[]
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param array $options {
+     *      Configuration options.
+     *
+     *      @type string $projectId The id of the project this trace belongs to.
+     *      @type string $traceId [optional] The trace id for this trace. 128-bit numeric
+     *            formatted as a 32-byte hex string. If not provided, one will be generated
+     *            automatically for you.
+     *      @type array $spans [optional] List of span data to load.
+     * }
+     */
+    public function __construct(array $options)
+    {
+        $this->projectId = $options['projectId'];
```

Comment:
I will do that for now, but I was debating attempting to extract the projectId out of the Trace record and injecting it into the requests before it's sent to the server. Trace seems like one of the few places where the projectId exists as a field in the record rather than just in the path to the resource, so I didn't see any other examples to follow.